SplitScreenScanner/Classes/SplitScannerCoordinator.swift

```diff
@@ -184,7 +196,17 @@ extension SplitScannerCoordinator: BarcodeScannerViewModelDelegate {
     func didScanBarcode(_ barcodeScannerViewModel: BarcodeScannerViewModel, barcode: String) {
         if currentlyDisplayedInfoVC is ScanHistoryTableViewController {
             guard let scanResult = delegate?.didScanBarcode(self, barcode: barcode) else { return }
+
             scanHistoryViewModel?.didScan(barcode: barcode, with: scanResult)
+
+            if delegate?.shouldDismiss(after: scanResult) == true {
+                rootCoordinator?.popCoordinator(self)
+
+                DispatchQueue.main.async { [weak self] in
+                    guard let sSelf = self else { return }
+                    sSelf.delegate?.didTriggerDismissalWithScan(sSelf)
```

Comment:
`sSelf` → `self`, or don’t use `[weak self]`. I’m not sure it makes sense to pop the coordinator, dispatch, then dismiss. That feels like it could fail—you still want to trigger dismissal even if the coordinator is deallocated. Does this makes sense? (including Jeremy’s suggestion above about `guard let delegate = delegate` at the start)
```swift
if delegate.shouldDismiss(after: scanResult) {
    DispatchQueue.main.async {
        rootCoordinator?.popCoordinator(self)
        self.delegate?.didTriggerDismissal(self, with: scanResult)
    }
}
```